src/edu/pugetsound/mathcs/architecture_nlp/features/spf/Interactor.java

```diff
@@ -0,0 +1,214 @@
+ * Copyright (C) 2011 - 2015 Yoav Artzi, All rights reserved.
+ * <p>
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ * <p>
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *******************************************************************************/
+package edu.pugetsound.mathcs.architecture_nlp.features.spf;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.cornell.cs.nlp.spf.base.hashvector.IHashVector;
+import edu.cornell.cs.nlp.spf.ccg.lexicon.LexicalEntry;
+import edu.cornell.cs.nlp.spf.data.IDataItem;
+import edu.cornell.cs.nlp.spf.data.ILabeledDataItem;
+import edu.cornell.cs.nlp.spf.data.collection.IDataCollection;
+import edu.cornell.cs.nlp.spf.data.sentence.Sentence;
+import edu.cornell.cs.nlp.spf.data.sentence.SentenceCollection;
+import edu.cornell.cs.nlp.spf.explat.IResourceRepository;
+import edu.cornell.cs.nlp.spf.explat.ParameterizedExperiment.Parameters;
+import edu.cornell.cs.nlp.spf.explat.resources.IResourceObjectCreator;
+import edu.cornell.cs.nlp.spf.explat.resources.usage.ResourceUsage;
+import edu.cornell.cs.nlp.spf.parser.IDerivation;
+import edu.cornell.cs.nlp.spf.parser.IOutputLogger;
+import edu.cornell.cs.nlp.spf.parser.IParser;
+import edu.cornell.cs.nlp.spf.parser.IParserOutput;
+import edu.cornell.cs.nlp.spf.parser.ccg.IWeightedParseStep;
+import edu.cornell.cs.nlp.spf.parser.ccg.cky.CKYDerivation;
+import edu.cornell.cs.nlp.spf.parser.ccg.model.IDataItemModel;
+import edu.cornell.cs.nlp.spf.parser.ccg.model.IModelImmutable;
+import edu.cornell.cs.nlp.spf.test.stats.ITestingStatistics;
+import edu.cornell.cs.nlp.utils.collections.ListUtils;
+import edu.cornell.cs.nlp.utils.filter.IFilter;
+import edu.cornell.cs.nlp.utils.log.ILogger;
+import edu.cornell.cs.nlp.utils.log.LoggerFactory;
+
+/**
+ * Module to allow the interaction of the Model with both singular sentences and with a set of sentences.
+ * When given a set of sentences, resulting parses are written out to file. This file is modeled off of Yoav 
+ * Artiz's Tester unit. Instead of taking in results from parses and catologuing results, this unit takes in
+ * sentences and returns LogicalRepresentation parses to be used by the rest of the system.
```

Comment:
Add a paragraph below this comment describing how this code differs from the corresponding class in the SPF jar file